File: avancado-intermediario/Python/Q5/personagem.py
from objeto import Objeto
from movimento import Movimento

class Personagem(Objeto, Movimento):

    def __init__(self, nome, x = 0.0, y = 0.0):
        super().__init__(x, y)
        self.__nome = nome

    def getNome(self):
        return self.__nome

    def mover(self, x, y):
        self.setX(x)
        self.setY(y)

    def __str__(self):
        return (f"{super().__str__()}"
                f"Nome = {self.getNome()}")

File: avancado-intermediario/Java/Q5/Personagem.java
public class Personagem extends Objeto implements Movimento {

    private String nome;

    public Personagem(String nome) {
        this(nome, 0.0, 0.0);
    }

    public Personagem(String nome, double x, double y) {
        super(x, y);
        this.nome = nome;
    }

    public String getNome() {
        return nome;
    }

    @Override
    public void mover(double x, double y) {
        setX(x);
        setY(y);
    }

    @Override
    public String toString() {
        return super.toString() +
               "Nome = " + nome;
    }
}
